### PR TITLE
Adding aws sync to s3 website bucket

### PR DESCRIPTION
**Pull request by Tatskaari**
First step in migrating please.build from our LAMP + cloudflare stack to an aws stack.

### CODE REVIEW

File: docs/BUILD [hunk 1]
```diff
@@ -82,6 +82,14 @@ tarball(
     labels = ["hlink:plz-out/pkg"],
 )
 
+tarball(
```
Comment (Tatskaari) [inline docs/BUILD:85]: The flat tarball was harder to deal with and was flattening the images into the web root for some reason.

File: .circleci/config.yml [hunk 2]
```diff
@@ -270,6 +271,16 @@ jobs:
...
+   aws-sync:
+     docker:
+       - image: amazon/aws-cli
+     steps:
+       - attach_workspace:
+           at: /tmp/site
+       - run:
+           name: Upload docs
+           working_directory: /tmp/site
+           command: tar -xzf deep-docs.tar.gz && aws s3 sync docs s3://please-build && aws s3 sync linux_amd64 s3://please-build/linux_amd64 && aws s3 sync darwin_amd64 s3://please-build/darwin_amd64 && aws s3 sync freebsd_amd64 s3://please-build/freebsd_amd64
```
Comment (peterebden) [inline .circleci/config.yml:283]: You don't have all the GPG signatures etc here, those are generated in the release step

File: .circleci/config.yml [hunk 3]
```diff
@@ -309,3 +320,9 @@ workflows:
           filters:
             branches:
               only: master
+      - aws-sync:
+          requires:
+            - build-linux
```
Comment (peterebden) [inline .circleci/config.yml:325]: this would require the darwin and freebsd builds as well, no?

File: .circleci/config.yml (hunk 2 above)
Comment (Tatskaari) [reply .circleci/config.yml:283]: Ah right I see... I guess the current server just routes requests so it doesn't need to worry about that. Probably needs some thought. I guess I'll come back to the releases. We can get the docs updating first.

File: .circleci/config.yml (hunk 3 above)
Comment (Tatskaari) [reply .circleci/config.yml:325]: Ah good catch! I've made this PR only implement syncing for docs now though.

**Review by peterebden — APPROVED** [review]
We'll need to sort out credentials in CircleCI presumably?

**Comment by Tatskaari** [comment]
> We'll need to sort out credentials in CircleCI presumably?

I _think_ I've already done so (set some env vars as the docs suggest)

**Pull request merged**